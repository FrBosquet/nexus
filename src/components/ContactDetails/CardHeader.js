import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { fromTheme } from '../../utils/styled';

const HeaderImage = styled('img')``;
const HeaderTitle = styled('h1')``;

const CardHeader = ({ className, name, picture }) => (
  <div className={className}>
    <HeaderTitle>{name}</HeaderTitle>
    <HeaderImage src={picture} />
  </div>
);

CardHeader.propTypes = {
  className: PropTypes.string,
  name: PropTypes.string.isRequired,
  picture: PropTypes.string.isRequired,
};

CardHeader.defaultProps = {
  className: '',
};

export default styled(CardHeader)`
  ${HeaderImage} {
    border-bottom: 1px solid ${fromTheme('--color-primary')};
    height: 32rem;
    object-fit: cover;
    width: 100%;
  }

  ${HeaderTitle} {
    color: white;
    padding: 1.5rem;
    position: absolute;
    text-shadow: 1px 1px black;
    margin-top: 26rem;
  }

  @media (${fromTheme('--screen-medium')}) {
    ${HeaderImage} {
      height: 15rem;
    }

    ${HeaderTitle} {
      margin-top: 10rem;
    }
  }
`;
