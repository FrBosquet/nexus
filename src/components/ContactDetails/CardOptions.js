import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const CardOptions = ({ className, children }) => (
  <div className={className}>{children}</div>
);

CardOptions.propTypes = {
  className: PropTypes.string,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]),
};

CardOptions.defaultProps = {
  className: '',
  children: null,
};

export default styled(CardOptions)`
  padding: 2rem;
  display: flex;
  justify-content: flex-end;

  *:not(:last-child) {
    margin-right: 1rem;
  }
`;
