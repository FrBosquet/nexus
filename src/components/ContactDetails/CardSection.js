import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import Icon from '../Common/Icon';
import { fromTheme } from '../../utils/styled';

const SectionContent = styled('div')``;

const CardSection = ({ className, children, icon }) => (
  <section className={className}>
    <Icon>{icon}</Icon>
    <SectionContent>{children}</SectionContent>
  </section>
);

CardSection.propTypes = {
  className: PropTypes.string,
  icon: PropTypes.string,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]),
};

CardSection.defaultProps = {
  className: '',
  icon: 'phone',
  children: null,
};

export default styled(CardSection)`
  display: flex;
  flex-direction: row;
  padding: 1rem 0;

  ${Icon} {
    margin: 0 2rem;
    color: ${fromTheme('--color-primary')};
  }

  ${SectionContent} {
    flex-grow: 1;
    padding-right: 2rem;
  }
`;
