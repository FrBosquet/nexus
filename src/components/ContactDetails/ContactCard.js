import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import CardSection from './CardSection';

const Header = styled('div')``;
const HeaderImage = styled('img')``;
const HeaderTitle = styled('h1')``;
const SectionItem = styled('div')``;
const SectionTitle = styled('h4')``;
const SectionSubtitle = styled('h5')``;

const ContactCard = ({
  className,
  name: { first, last },
  picture: { large: picture },
  phone,
  cell,
  email,
  location: { street, postcode, city },
}) => (
  <article className={className}>
    <Header>
      <HeaderTitle>{`${first} ${last}`}</HeaderTitle>
      <HeaderImage src={picture} />
    </Header>
    <CardSection icon="phone">
      {cell && (
        <SectionItem>
          <SectionTitle>{cell}</SectionTitle>
          <SectionSubtitle>Cell</SectionSubtitle>
        </SectionItem>
      )}
      {phone && (
        <SectionItem>
          <SectionTitle>{phone}</SectionTitle>
          <SectionSubtitle>Home</SectionSubtitle>
        </SectionItem>
      )}
    </CardSection>
    <CardSection icon="email">
      <SectionItem>
        <SectionTitle>{email}</SectionTitle>
        <SectionSubtitle>Personal</SectionSubtitle>
      </SectionItem>
    </CardSection>
    <CardSection icon="home">
      <SectionItem>
        <SectionTitle>{`${street}, ${postcode} ${city}`}</SectionTitle>
        <SectionSubtitle>Home</SectionSubtitle>
      </SectionItem>
    </CardSection>
  </article>
);

ContactCard.propTypes = {
  className: PropTypes.string,
  name: PropTypes.shape({
    first: PropTypes.string.isRequired,
    last: PropTypes.string,
  }).isRequired,
  picture: PropTypes.shape({
    large: PropTypes.string.isRequired,
  }).isRequired,
  phone: PropTypes.string,
  cell: PropTypes.string.isRequired,
  email: PropTypes.string,
  location: PropTypes.shape({
    street: PropTypes.string,
    postcode: PropTypes.number,
    city: PropTypes.string,
  }).isRequired,
};

ContactCard.defaultProps = {
  className: '',
  phone: '',
  email: '',
};

export default styled(ContactCard)`
  width: 100%;

  animation-name: fade;
  animation-duration: 0.5s;

  z-index: 101;

  ${Header} {
    ${HeaderImage} {
      border-bottom: 1px solid ${props => props.theme['--color-primary']};
      height: 32rem;
      object-fit: cover;
      width: 100%;
    }

    ${HeaderTitle} {
      color: white;
      padding: 1.5rem;
      position: absolute;
      text-shadow: 1px 1px black;
      margin-top: 26rem;
    }
  }

  ${CardSection} {
    ${SectionTitle} {
      color: ${props => props.theme['--color-primary']};
      text-align: right;
    }

    ${SectionSubtitle} {
      color: ${props => props.theme['--color-secondary']};
      text-align: right;
    }

    &:not(:last-child) {
      border-bottom: 1px solid ${props => props.theme['--color-secondary']};
    }
  }

  @keyframes fade {
    from {
      opacity: 0;
      transform: translateY(100px);
    }

    to {
      opacity: 100;
      transform: translateY(0px);
    }
  }

  @media (${props => props.theme['--screen-medium']}) {
    border: 1px solid ${props => props.theme['--color-primary']};
    box-shadow: 0px 0px 50px 0px ${props => props.theme['--color-dark']};
    margin: auto;
    width: 46rem;

    ${Header} {
      ${HeaderImage} {
        height: 15rem;
      }

      ${HeaderTitle} {
        margin-top: 10rem;
      }
    }
  }
`;
