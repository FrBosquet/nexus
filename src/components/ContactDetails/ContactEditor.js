import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import RoundButton from '../Common/RoundButton';

import CardSection from './CardSection';
import { fromTheme } from '../../utils/styled';
import CardHeader from './CardHeader';
import CardOptions from './CardOptions';

const SectionSubtitle = styled('h5')``;
const Input = styled('input')``;

class ContactEditor extends Component {
  static propTypes = {
    className: PropTypes.string,
    name: PropTypes.shape({
      first: PropTypes.string.isRequired,
      last: PropTypes.string,
    }).isRequired,
    picture: PropTypes.shape({
      large: PropTypes.string.isRequired,
    }).isRequired,
    phone: PropTypes.string,
    cell: PropTypes.string.isRequired,
    email: PropTypes.string,
    location: PropTypes.shape({
      street: PropTypes.string,
      postcode: PropTypes.number,
      city: PropTypes.string,
    }).isRequired,
    handleSave: PropTypes.func.isRequired,
    handleClose: PropTypes.func.isRequired,
  };

  static defaultProps = {
    className: '',
    phone: '',
    email: '',
  };

  constructor(props) {
    super(props);
    const {
      name: { first, last },
      picture: { large: picture },
      phone,
      cell,
      email,
      location: { street, postcode, city },
    } = props;

    this.state = {
      firstName: first,
      lastName: last,
      picture,
      phone,
      cell,
      email,
      street,
      postcode,
      city,
    };
  }

  makeHandleOnChange = field => event =>
    this.setState({ [field]: event.target.value });

  onSave = () => {
    const { handleClose, handleSave } = this.props;
    handleSave(this.state);
    handleClose();
  };

  render() {
    const { className, handleClose } = this.props;
    const {
      firstName,
      lastName,
      picture,
      phone,
      cell,
      email,
      street,
      postcode,
      city,
    } = this.state;

    const { makeHandleOnChange } = this;
    return (
      <article className={className}>
        <CardHeader name={`${firstName} ${lastName}`} picture={picture} />
        <CardSection icon="person">
          <Input onChange={makeHandleOnChange('firstName')} value={firstName} />
          <SectionSubtitle>First name</SectionSubtitle>
          <Input onChange={makeHandleOnChange('lastName')} value={lastName} />
          <SectionSubtitle>Last name</SectionSubtitle>
          <Input onChange={makeHandleOnChange('picture')} value={picture} />
          <SectionSubtitle>Picture url</SectionSubtitle>
        </CardSection>
        <CardSection icon="phone">
          <Input onChange={makeHandleOnChange('cell')} value={cell} />
          <SectionSubtitle>Cell</SectionSubtitle>
          <Input onChange={makeHandleOnChange('phone')} value={phone} />
          <SectionSubtitle>Phone</SectionSubtitle>
        </CardSection>
        <CardSection icon="email">
          <Input onChange={makeHandleOnChange('email')} value={email} />
          <SectionSubtitle>email</SectionSubtitle>
        </CardSection>
        <CardSection icon="home">
          <Input onChange={makeHandleOnChange('street')} value={street} />
          <SectionSubtitle>Street</SectionSubtitle>
          <Input
            onChange={makeHandleOnChange('postcode')}
            type="number"
            value={postcode}
          />
          <SectionSubtitle>Postcode</SectionSubtitle>
          <Input onChange={makeHandleOnChange('city')} value={city} />
          <SectionSubtitle>City</SectionSubtitle>
        </CardSection>
        <CardOptions>
          <RoundButton onClick={this.onSave} icon="save" />
          <RoundButton onClick={handleClose} icon="close" />
        </CardOptions>
      </article>
    );
  }
}

export default styled(ContactEditor)`
  width: 100%;

  animation-name: fade;
  animation-duration: 0.5s;

  z-index: 101;

  ${CardSection} {
    ${Input} {
      color: ${fromTheme('--color-primary')};
      text-align: right;
      width: 100%;
      border: none;
      border-bottom: 1px solid ${fromTheme('--color-blueish-white')};
      transition: padding 0.5s;
      font-weight: 800;

      &:focus {
        padding: 0.5rem;
        outline: none;
        color: ${fromTheme('--color-main')};
        border-bottom: 1px solid ${fromTheme('--color-main')};
      }
    }

    &:not(:last-child) {
      border-bottom: 1px solid ${fromTheme('--color-secondary')};
    }
  }

  @keyframes fade {
    from {
      opacity: 0;
      transform: translateY(100px);
    }

    to {
      opacity: 100;
      transform: translateY(0px);
    }
  }

  @media (${fromTheme('--screen-medium')}) {
    border: 1px solid ${fromTheme('--color-primary')};
    box-shadow: 0px 0px 50px 0px ${fromTheme('--color-dark')};
    margin: auto;
    width: 46rem;
  }
`;
