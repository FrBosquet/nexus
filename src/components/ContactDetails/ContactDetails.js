import React, { Component } from 'react';
import { rgba } from 'polished';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import Icon from '../Common/Icon';
import Link from '../Common/Link';
import Spinner from '../Common/Spinner';

import ContactCard from './ContactCard';

import Contacts from '../../services/contacts';
import { fromTheme } from '../../utils/styled';

const Container = styled('section')``;
const Header = styled('header')``;

class ContactDetails extends Component {
  static defaultProps = {
    className: '',
  };

  static propTypes = {
    className: PropTypes.string,
    match: PropTypes.shape({
      params: PropTypes.shape({
        id: PropTypes.string.isRequired,
      }).isRequired,
    }).isRequired,
  };

  state = {
    contact: null,
  };

  componentDidMount() {
    const {
      match: {
        params: { id },
      },
    } = this.props;
    this.refreshContact(id);
  }

  componentDidUpdate(prevProps) {
    const {
      match: {
        params: { id: currentId },
      },
    } = this.props;
    const {
      match: {
        params: { id: prevId },
      },
    } = prevProps;

    if (currentId !== prevId) this.refreshContact(currentId);
  }

  async refreshContact(id) {
    this.setState({ contact: null });
    const contact = await Contacts.read(id);
    this.setState({ contact });
  }

  render() {
    const { className } = this.props;
    const { contact } = this.state;

    return (
      <article className={className}>
        <Header>
          <Link to="/">
            <Icon>arrow_back_ios</Icon>
          </Link>
        </Header>
        {contact ? (
          <ContactCard {...contact} />
        ) : (
          <Container>
            <Spinner />
          </Container>
        )}
      </article>
    );
  }
}

export default styled(ContactDetails)`
  background: ${fromTheme('--color-light')};
  height: calc(100% - 2.5rem);
  position: fixed;
  top: 2.5rem;
  width: 100%;
  z-index: 101;
  //
  ${Header} {
    position: absolute;

    ${fromTheme('--font-extra-large')};
    align-items: center;
    display: flex;
    height: 5rem;
    justify-content: center;
    text-align: center;

    ${Icon} {
      height: 5rem;
      left: 0;
      line-height: 5rem;
      position: absolute;
      top: 0;
      width: 5rem;
    }
  }

  ${Container} {
    display: flex;
    align-items: center;
    justify-content: center;
  }

  @media (${fromTheme('--screen-medium')}) {
    border-left: 1px solid ${props => rgba(props.theme['--color-dark'], 0.1)};
    left: 32rem;
    width: calc(100% - 32rem);

    ${Header} {
      position: relative;
    }
  }
`;
