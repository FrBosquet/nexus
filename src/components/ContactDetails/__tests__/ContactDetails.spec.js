import React from 'react';
import { render, cleanup, waitForElement } from 'react-testing-library';

import ContactDetails from '../ContactDetails';
import withProvider from '../../../utils/hocs';

import Contact from '../../../services/contacts';

jest.mock('../../../services/contacts');

describe('Contact details', () => {
  const props = { match: { params: { id: 'id1' } } };
  const response = {
    name: { first: 'Fran', last: 'Bosquet' },
    picture: { large: 'url' },
    cell: '555-555-555',
    location: { street: 'Fake Street 123' },
  };
  const Enhanced = withProvider(ContactDetails);

  beforeEach(() => {
    Contact.read.mockClear();
  });

  afterEach(() => {
    cleanup();
  });

  it('renders by default', () => {
    const { container } = render(<Enhanced {...props} />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('calls Contact.read on mount', () => {
    render(<Enhanced {...props} />);
    expect(Contact.read).toHaveBeenCalledTimes(1);
    expect(Contact.read).toHaveBeenCalledWith('id1');
  });

  it('calls Contact.read on props change', () => {
    const { rerender } = render(<Enhanced {...props} />);
    const newProps = { match: { params: { id: 'id2' } } };
    rerender(<Enhanced {...newProps} />);
    expect(Contact.read).toHaveBeenCalledTimes(2);
    expect(Contact.read).toHaveBeenCalledWith('id2');
  });

  it('dont call Contact.read on props change when ids are the same', () => {
    const { rerender } = render(<Enhanced {...props} />);
    const newProps = { match: { params: { id: 'id1' } } };
    rerender(<Enhanced {...newProps} />);
    expect(Contact.read).toHaveBeenCalledTimes(1);
  });

  it('renders a Spinner while waiting from service data', () => {
    const { getByTestId } = render(<Enhanced {...props} />);
    expect(getByTestId('spinner')).toBeDefined();
  });

  it('renders a ContactCard with the provided contact data', async () => {
    Contact.read.mockReturnValueOnce(response);
    const { getByText } = render(<Enhanced {...props} />);
    await waitForElement(() => getByText('Fran Bosquet'));
    expect(getByText('Fran Bosquet')).toBeDefined();
  });

  describe('contact editor', () => {
    beforeEach(() => {
      Contact.read.mockReturnValueOnce(response);
    });

    it('renders contact editor', async () => {
      const { getByText, container } = render(<Enhanced {...props} />);
      await waitForElement(() => getByText('Fran Bosquet'));
      getByText('edit').click();
      expect(container).toMatchSnapshot('editor');
      expect(getByText('save')).toBeDefined();
      expect(getByText('close')).toBeDefined();
    });

    it('updates a contact on save', async () => {
      const { getByText, container } = render(<Enhanced {...props} />);
      await waitForElement(() => getByText('Fran Bosquet'));
      getByText('edit').click();
      getByText('save').click();
      expect(Contact.update).toBeCalled();
      expect(container).toMatchSnapshot('detail');
    });

    it('cancels edition', async () => {
      const { getByText, container } = render(<Enhanced {...props} />);
      await waitForElement(() => getByText('Fran Bosquet'));
      getByText('edit').click();
      getByText('close').click();
      expect(getByText('edit')).toBeDefined();
      expect(container).toMatchSnapshot('detail');
    });
  });
});
