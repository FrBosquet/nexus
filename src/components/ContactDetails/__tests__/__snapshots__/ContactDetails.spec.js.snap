// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact details contact editor cancels edition: detail 1`] = `
<div>
  <article
    class="sc-jzJRlG cfNkLn"
  >
    <header
      class="sc-fjdhpX nsimD"
    >
      <a
        class=""
        href="/"
      >
        <i
          class="sc-bdVaJa bDWFJH material-icons"
        >
          arrow_back_ios
        </i>
      </a>
    </header>
    <article
      class="sc-iwsKbI bqZdYh"
    >
      <div
        class="sc-bZQynM daovDU"
      >
        <h1
          class="sc-EHOje jvzENE"
        >
          Fran Bosquet
        </h1>
        <img
          class="sc-ifAKCX cFlEyZ"
          src="url"
        />
      </div>
      <section
        class="sc-bxivhb iMGPqq"
      >
        <i
          class="sc-bdVaJa bDWFJH material-icons"
        >
          phone
        </i>
        <div
          class="sc-htpNat KtFsv"
        >
          <h4
            class="sc-htoDjs bZzhDu"
          >
            555-555-555
          </h4>
          <h5
            class="sc-dnqmqq bDsesi"
          >
            Cell
          </h5>
          
        </div>
      </section>
      <section
        class="sc-bxivhb iMGPqq"
      >
        <i
          class="sc-bdVaJa bDWFJH material-icons"
        >
          email
        </i>
        <div
          class="sc-htpNat KtFsv"
        >
          <h4
            class="sc-htoDjs bZzhDu"
          />
          <h5
            class="sc-dnqmqq bDsesi"
          >
            Personal
          </h5>
        </div>
      </section>
      <section
        class="sc-bxivhb iMGPqq"
      >
        <i
          class="sc-bdVaJa bDWFJH material-icons"
        >
          home
        </i>
        <div
          class="sc-htpNat KtFsv"
        >
          <h4
            class="sc-htoDjs bZzhDu"
          >
            Fake Street 123, undefined undefined
          </h4>
          <h5
            class="sc-dnqmqq bDsesi"
          >
            Home
          </h5>
        </div>
      </section>
      <div
        class="sc-gzVnrw hFoRxL"
      >
        <button
          class="sc-bwzfXH fUeTHc"
          type="button"
        >
          <i
            class="sc-bdVaJa bDWFJH material-icons"
          >
            edit
          </i>
        </button>
      </div>
    </article>
  </article>
</div>
`;

exports[`Contact details contact editor renders contact editor: editor 1`] = `
<div>
  <article
    class="sc-jzJRlG cfNkLn"
  >
    <header
      class="sc-fjdhpX nsimD"
    >
      <a
        class=""
        href="/"
      >
        <i
          class="sc-bdVaJa bDWFJH material-icons"
        >
          arrow_back_ios
        </i>
      </a>
    </header>
    <article
      class="sc-VigVT juUpIN"
    >
      <div
        class="sc-bZQynM daovDU"
      >
        <h1
          class="sc-EHOje jvzENE"
        >
          Fran Bosquet
        </h1>
        <img
          class="sc-ifAKCX cFlEyZ"
          src="url"
        />
      </div>
      <section
        class="sc-bxivhb iMGPqq"
      >
        <i
          class="sc-bdVaJa bDWFJH material-icons"
        >
          person
        </i>
        <div
          class="sc-htpNat KtFsv"
        >
          <input
            class="sc-gqjmRU hnZXkc"
            value="Fran"
          />
          <h5
            class="sc-gZMcBi giNnHm"
          >
            First name
          </h5>
          <input
            class="sc-gqjmRU hnZXkc"
            value="Bosquet"
          />
          <h5
            class="sc-gZMcBi giNnHm"
          >
            Last name
          </h5>
          <input
            class="sc-gqjmRU hnZXkc"
            value="url"
          />
          <h5
            class="sc-gZMcBi giNnHm"
          >
            Picture url
          </h5>
        </div>
      </section>
      <section
        class="sc-bxivhb iMGPqq"
      >
        <i
          class="sc-bdVaJa bDWFJH material-icons"
        >
          phone
        </i>
        <div
          class="sc-htpNat KtFsv"
        >
          <input
            class="sc-gqjmRU hnZXkc"
            value="555-555-555"
          />
          <h5
            class="sc-gZMcBi giNnHm"
          >
            Cell
          </h5>
          <input
            class="sc-gqjmRU hnZXkc"
            value=""
          />
          <h5
            class="sc-gZMcBi giNnHm"
          >
            Phone
          </h5>
        </div>
      </section>
      <section
        class="sc-bxivhb iMGPqq"
      >
        <i
          class="sc-bdVaJa bDWFJH material-icons"
        >
          email
        </i>
        <div
          class="sc-htpNat KtFsv"
        >
          <input
            class="sc-gqjmRU hnZXkc"
            value=""
          />
          <h5
            class="sc-gZMcBi giNnHm"
          >
            email
          </h5>
        </div>
      </section>
      <section
        class="sc-bxivhb iMGPqq"
      >
        <i
          class="sc-bdVaJa bDWFJH material-icons"
        >
          home
        </i>
        <div
          class="sc-htpNat KtFsv"
        >
          <input
            class="sc-gqjmRU hnZXkc"
            value="Fake Street 123"
          />
          <h5
            class="sc-gZMcBi giNnHm"
          >
            Street
          </h5>
          <input
            class="sc-gqjmRU hnZXkc"
            type="number"
            value=""
          />
          <h5
            class="sc-gZMcBi giNnHm"
          >
            Postcode
          </h5>
          <input
            class="sc-gqjmRU hnZXkc"
            value=""
          />
          <h5
            class="sc-gZMcBi giNnHm"
          >
            City
          </h5>
        </div>
      </section>
      <div
        class="sc-gzVnrw hFoRxL"
      >
        <button
          class="sc-bwzfXH fUeTHc"
          type="button"
        >
          <i
            class="sc-bdVaJa bDWFJH material-icons"
          >
            save
          </i>
        </button>
        <button
          class="sc-bwzfXH fUeTHc"
          type="button"
        >
          <i
            class="sc-bdVaJa bDWFJH material-icons"
          >
            close
          </i>
        </button>
      </div>
    </article>
  </article>
</div>
`;

exports[`Contact details contact editor updates a contact on save: detail 1`] = `
<div>
  <article
    class="sc-jzJRlG cfNkLn"
  >
    <header
      class="sc-fjdhpX nsimD"
    >
      <a
        class=""
        href="/"
      >
        <i
          class="sc-bdVaJa bDWFJH material-icons"
        >
          arrow_back_ios
        </i>
      </a>
    </header>
    <section
      class="sc-jTzLTM eCMmlJ"
    >
      <svg
        class="lds-infinity"
        data-testid="spinner"
        height="100px"
        preserveAspectRatio="xMidYMid"
        viewBox="0 0 100 100"
        width="100px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M24.3,30C11.4,30,5,43.3,5,50s6.4,20,19.3,20c19.3,0,32.1-40,51.4-40 C88.6,30,95,43.3,95,50s-6.4,20-19.3,20C56.4,70,43.6,30,24.3,30z"
          fill="none"
          ng-attr-stroke="{{config.stroke}}"
          ng-attr-stroke-dasharray="{{config.dasharray}}"
          ng-attr-stroke-width="{{config.width}}"
          stroke="#0084ff"
          stroke-dasharray="5.131778564453125 5.131778564453125"
          stroke-width="8"
        >
          <animate
            attributeName="stroke-dashoffset"
            begin="0s"
            calcMode="linear"
            dur="7.8"
            keyTimes="0;1"
            repeatCount="indefinite"
            values="0;256.58892822265625"
          />
        </path>
      </svg>
    </section>
  </article>
</div>
`;

exports[`Contact details renders by default 1`] = `
<article
  class="sc-jzJRlG cfNkLn"
>
  <header
    class="sc-fjdhpX nsimD"
  >
    <a
      class=""
      href="/"
    >
      <i
        class="sc-bdVaJa bDWFJH material-icons"
      >
        arrow_back_ios
      </i>
    </a>
  </header>
  <section
    class="sc-jTzLTM eCMmlJ"
  >
    <svg
      class="lds-infinity"
      data-testid="spinner"
      height="100px"
      preserveAspectRatio="xMidYMid"
      viewBox="0 0 100 100"
      width="100px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M24.3,30C11.4,30,5,43.3,5,50s6.4,20,19.3,20c19.3,0,32.1-40,51.4-40 C88.6,30,95,43.3,95,50s-6.4,20-19.3,20C56.4,70,43.6,30,24.3,30z"
        fill="none"
        ng-attr-stroke="{{config.stroke}}"
        ng-attr-stroke-dasharray="{{config.dasharray}}"
        ng-attr-stroke-width="{{config.width}}"
        stroke="#0084ff"
        stroke-dasharray="5.131778564453125 5.131778564453125"
        stroke-width="8"
      >
        <animate
          attributeName="stroke-dashoffset"
          begin="0s"
          calcMode="linear"
          dur="7.8"
          keyTimes="0;1"
          repeatCount="indefinite"
          values="0;256.58892822265625"
        />
      </path>
    </svg>
  </section>
</article>
`;
